cluster.go

```diff
@@ -100,6 +112,8 @@ type Cluster struct {
 	// used to deduplicate calls to sync
 	syncDedupe *dedupe
 
+	down uint64
```

Comment:
I used a atomic variable for two reasons:

1. It was quick to implement
2. Since this is potentially checked on every command invocation the check must not be slow especially under contention. As long as the value of the variable does not change all reads should be perfectly scalable.

I have no problem with replacing this by something else. Also I just realized that this is missing a big fat warning that this is accessed using the `sync/atomic` package...